quipucords/api/source/view.py

```diff
@@ -39,12 +39,44 @@ def expand_credential(json_source):
     Take source object with credential id and pull object from db.
     create slim dictionary version of the host credential with name an value
     to return to user.
+
+    :param json_source: JSON source data from serializer
+    :returns: JSON data
     """
     cred_ids = json_source.get('credentials', [])
-    slim_cred = Credential.objects.filter(pk__in=cred_ids).values('id', 'name')
+    slim_cred = Credential.objects.filter(
+        pk__in=cred_ids).values('id', 'name')
     # Update source JSON with cred JSON
     if slim_cred:
         json_source[CREDENTIALS_KEY] = slim_cred
+    return json_source
+
+
+def format_source(json_source):
+    """Format source with credentials and most recent connection scan.
+
+    :param json_source: JSON source data from serializer
+    :returns: JSON data
+    """
+    json_source = expand_credential(json_source)
+    source_id = json_source.get('id')
+
+    scan_task_qs = ScanTask.objects.filter(
+        source__id=source_id,
+        scan_type=ScanTask.SCAN_TYPE_CONNECT)
+    if scan_task_qs:
```

Comment:
I think this will not be null if if the results are empty.  You may want to move scan_task_qs.latest('start_time') above this check.

Reply:
put in a guard with `count() > 0`